### PR TITLE
shutdown: Select respective action on halt and soft-reboot utilities

### PR DESCRIPTION
**Pull request by mobin-2008**
The #333 came to my mind and I checked the `shutdown.cc` code to make sure that's OK and It's not.

`Signed-off-by: Mobin Aydinfar <mobin@mobintestserver.ir>`

### CODE REVIEW

File: src/shutdown.cc [hunk 1]
```diff
@@ -257,9 +259,15 @@ int main(int argc, char **argv)
     auto shutdown_type = shutdown_type_t::POWEROFF;
 
     const char *execname = base_name(argv[0]);
-    if (strcmp(execname, reboot_execname) == 0) {
+    if (strcmp(execname, halt_execname) == 0) {
+        shutdown_type = shutdown_type_t::HALT;
+    }
+    else if (strcmp(execname, reboot_execname) == 0) {
```
Comment (davmac314) [inline src/shutdown.cc:265]: From the manual page:

       The default action is to power down the system if called as either
       "shutdown", "halt", or "poweroff"

This breaks that.

File: src/shutdown.cc [hunk 2]
```diff
@@ -257,9 +259,15 @@ int main(int argc, char **argv)
...
         shutdown_type = shutdown_type_t::REBOOT;
     }
+    else if (strcmp(execname, soft_reboot_execname) == 0) {
+        shutdown_type = shutdown_type_t::SOFTREBOOT;
+    }
```
Comment (davmac314) [inline src/shutdown.cc:270]: This is ok but the manual page needs updating.

**Review by davmac314 — CHANGES_REQUESTED** [review]

File: src/shutdown.cc (hunk 1 above)
Comment (mobin-2008) [reply src/shutdown.cc:265]: It's confusing for me that `halt` and `shutdown -h` have different effect but I'm  going to fix it to match the doc.

File: src/shutdown.cc (hunk 2 above)
Comment (mobin-2008) [reply src/shutdown.cc:270]: From shutdown.8:

> -s
> Restart the service manager and all user-space services without restarting the system.
> This is the default if executed as soft-reboot.

I don't see what should get updated in this doc.

Comment (davmac314) [reply src/shutdown.cc:270]: You're right, it's already ok. I was looking at the previous version man page by mistake.

File: src/shutdown.cc (hunk 1 above)
Comment (mobin-2008) [reply src/shutdown.cc]: The halt check is removed

**Review by davmac314 — APPROVED** [review]

**Comment by mobin-2008** [comment]
@davmac314 This is my last patch before the next release.

**Pull request merged**